server/src/internalClusterTest/java/org/opensearch/indices/replication/SearchReplicaRestoreIT.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnDocRep_RestoreOnDocRepWithSearchReplica() throws Exception {
+        bootstrapIndexWithOutSearchReplicas(ReplicationType.DOCUMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        SnapshotRestoreException exception = expectThrows(
+            SnapshotRestoreException.class,
+            () -> restoreSnapshot(
+                REPOSITORY_NAME,
+                SNAPSHOT_NAME,
+                INDEX_NAME,
+                RESTORED_INDEX_NAME,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT)
+                    .put(SETTING_NUMBER_OF_SEARCH_REPLICAS, 1)
+                    .build()
+            )
+        );
+        assertTrue(exception.getMessage().contains(getSnapshotExceptionMessage(ReplicationType.DOCUMENT, ReplicationType.DOCUMENT)));
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnDocRep_RestoreOnSegRep() throws Exception {
+        bootstrapIndexWithOutSearchReplicas(ReplicationType.DOCUMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        restoreSnapshot(
+            REPOSITORY_NAME,
+            SNAPSHOT_NAME,
+            INDEX_NAME,
+            RESTORED_INDEX_NAME,
+            Settings.builder().put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT).build()
+        );
+        ensureGreen(RESTORED_INDEX_NAME);
+        assertEquals(0, getNumberOfSearchReplicas(RESTORED_INDEX_NAME));
+
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnDocRep_RestoreOnSegRepWithSearchReplica() throws Exception {
+        bootstrapIndexWithOutSearchReplicas(ReplicationType.DOCUMENT);
+        createRepoAndSnapshot(REPOSITORY_NAME, FS_REPOSITORY_TYPE, SNAPSHOT_NAME, INDEX_NAME);
+
+        restoreSnapshot(
+            REPOSITORY_NAME,
+            SNAPSHOT_NAME,
+            INDEX_NAME,
+            RESTORED_INDEX_NAME,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+                .put(SETTING_NUMBER_OF_SEARCH_REPLICAS, 1)
+                .build()
+        );
+        ensureYellowAndNoInitializingShards(RESTORED_INDEX_NAME);
+        internalCluster().startDataOnlyNode();
+        ensureGreen(RESTORED_INDEX_NAME);
+        assertEquals(1, getNumberOfSearchReplicas(RESTORED_INDEX_NAME));
+
+        SearchResponse resp = client().prepareSearch(RESTORED_INDEX_NAME).setQuery(QueryBuilders.matchAllQuery()).get();
+        assertHitCount(resp, DOC_COUNT);
+    }
+
+    public void testSearchReplicaRestore_WhenSnapshotOnSegRep_RestoreOnDocRep() throws Exception {
```

Comment:
also don't need this case if the original index had no SR

Reply:
Sure